config/components.php

```diff
@@ -154,9 +154,9 @@
 			'words'     => false,
 		];
 
-		$collection  = clone $collection;
-		$options     = array_merge($defaults, $params);
-		$query       = trim($query);
+		$collection = clone $collection;
```

Comment:
I think this clone operation can be removed completely as both `$collection->limit()` (via `$collection->slice()`) and `$collection->filter()` also clone the collection before any operation. Removing this line would increase performance for large collections.

```suggestion
```